<?xml version="1.0"?>
<launch>
  <arg name="side" default="r"/> <!-- "b" = blue side, "r" = red side  -->
  <arg name="self_play" default="0"/>
  <arg name="batch_size" default="16"/>
  <arg name="capacity" default="1000"/>
  <arg name="episode" default="0"/>
  <arg name="gamma" default="0.99"/>
  <arg name="learning_rate" default="0.0005"/>
  <arg name="model_path" default="/tmp/model.pth"/>
  <arg name="memory_path" default="/tmp/memory.pth"/>

    <!-- You can change control program  -->

    <include file="$(find burger_navigation)/launch/burger_navigation.launch" />

    <group if="$(eval self_play==0)"> 
      <node pkg="burger_war_dev" type="reinforcement_operation.py" name="DQNRun" output="screen">
        <param name="side" value="$(arg side)"/>
        <param name="batch_size" value="$(arg batch_size)"/>
        <param name="capacity" value="$(arg capacity)"/>
        <param name="episode" value="$(arg episode)"/>
        <param name="gamma" value="$(arg gamma)"/>
        <param name="learning_rate" value="$(arg learning_rate)"/>
        <param name="model_path" value="$(arg model_path)"/>
        <param name="memory_path" value="$(arg memory_path)"/>
      </node>
    </group>

    <!-- <group if="$(eval self_play==1)"> --> 
    <!--   <node pkg="burger_war_dev" type="dqn_self_play.py" name="DQNRun" output="screen"> -->
    <!--     <param name="side" value="$(arg side)"/> -->
    <!--   </node> -->
    <!-- </group> -->

    <!-- End of your space  -->
</launch>
